routing.go

```diff
@@ -83,3 +84,30 @@ func nsToCid(ns string) (cid.Cid, error) {
 
 	return cid.NewCidV1(cid.Raw, h), nil
 }
+
+func NewDiscoveryRouting(disc discovery.Discovery, opts ...discovery.Option) *DiscoveryRouting {
+	return &DiscoveryRouting{disc, opts}
+}
+
+type DiscoveryRouting struct {
+	discovery.Discovery
+	opts []discovery.Option
+}
+
+func (r *DiscoveryRouting) Provide(ctx context.Context, c cid.Cid, bcast bool) error {
+	if !bcast {
+		return nil
+	}
+
+	_, err := r.Advertise(ctx, cidToNs(c), r.opts...)
+	return err
+}
+
+func (r *DiscoveryRouting) FindProvidersAsync(ctx context.Context, c cid.Cid, limit int) <-chan peer.AddrInfo {
+	ch, _ := r.FindPeers(ctx, cidToNs(c), append([]discovery.Option{discovery.Limit(limit)}, r.opts...)...)
+	return ch
+}
+
+func cidToNs(c cid.Cid) string {
+	return "/provider/" + c.String()
```

Comment:
Perhaps use `/cid` as the prefix?

Reply:
No strong preferences here. `/cid` is defining based on the key type (CID) and `/provider` is based on the value type (multiaddr/provider info).

My understanding is that we wanted to get away from advertising CIDs anyway (and instead advertise multihashes) in which case `/cid` probably isn't the way to go.

@Stebalien thoughts?

Reply:
let's go with `/cid`

Reply:
So, @aschmahmann is right. We'd like to move away from advertising CIDs to advertising hashes. Actually, what if we just went ahead and did `/content/$multihash` from the start? Or `/block/$multihash`?